snapcraft/plugins/python.py

```diff
@@ -438,17 +439,13 @@ def list(self, exec_func=None):
         """Return a dict of installed python packages with versions."""
         if not exec_func:
             exec_func = self._exec_func
-        cmd = [*self._runnable, 'list']
+        cmd = [*self._runnable, 'list', '--format=json']
 
         output = exec_func(cmd, env=self._env)
-        package_listing = {}
-        version_regex = re.compile('\((.+)\)')
-        for line in output.splitlines():
-            line = line.split()
-            m = version_regex.search(line[1])
-            package_listing[line[0]] = m.group(1)
-
-        return package_listing
+        return {
+            package['name']: package['version']
+            for package in json.loads(output)
+        }
```

Comment:
That makes sense. The way you put this PR in the context of the bug report made it seem to me as if it's addressing the bug, which isn't the case. It's just improving the parsing. A follow-up is going to be needed.